.github/workflows/e2e.yaml

```diff
@@ -33,9 +33,10 @@ jobs:
       run: |
         mkdir _output
 
-        docker build -t gcr.io/k8s-staging-kas-network-proxy/proxy-agent:master -f artifacts/images/agent-build.Dockerfile .
+        # TODO: match BASEIMAGE with Makefile default (nonroot variant)
+        docker build -t gcr.io/k8s-staging-kas-network-proxy/proxy-agent:master -f artifacts/images/agent-build.Dockerfile --build-arg GO_TOOLCHAIN=golang --build-arg GO_VERSION=${{ env.GO_VERSION }} --build-arg BASEIMAGE=gcr.io/distroless/static-debian11 .
```

Comment:
nit: consider making BASEIMAGE an env arg

Reply:
Done. (Did it for GO_TOOLCHAIN as well just for consistency.)